inline.h

```diff
@@ -57,6 +57,94 @@ Perl_av_count(pTHX_ AV *av)
     return AvFILL(av) + 1;
 }
 
+/* ------------------------------- av.c ------------------------------- */
+
+/*
+=for apidoc av_store_simple
+
+This is a cut-down version of av_store that assumes that the array is
+very straightforward - no magic, not readonly, and AvREAL - and that
+C<key> is not negative. This function MUST NOT be used in situations
+where any of those assumptions may not hold.
+
+Stores an SV in an array.  The array index is specified as C<key>.  The
+return value will be C<NULL> if the operation failed or if the value did not
+need to be actually stored within the array (as in the case of tied
+arrays).  Otherwise, it can be dereferenced
+to get the C<SV*> that was stored
+there (= C<val>)).
+
+Note that the caller is responsible for suitably incrementing the reference
+count of C<val> before the call, and decrementing it if the function
+returned C<NULL>.
+
+Approximate Perl equivalent: C<splice(@myarray, $key, 1, $val)>.
```

Comment:
Isn't it more like `$myarray[$key] = $val`? The behaviour beyond the end certainly seems so:
```
% perl -wle '@a=(1,2); $a[3]= 3; print join "-", @a'
Use of uninitialized value within @a in join or string at -e line 1.
1-2--3
% perl -wle '@a=(1,2); splice @a, 3, 1, 3; print join "-", @a'
splice() offset past end of array at -e line 1.
1-2-3
```

Reply:
`Approximate Perl equivalent: C<splice(@myarray, $key, 1, $val)>` is straight out of `av_store`'s documentation. Are you suggesting that instance should also be revised?

Reply:
> `Approximate Perl equivalent: C<splice(@myarray, $key, 1, $val)>` is straight out of `av_store`'s documentation. Are you suggesting that instance should also be revised?

Ah, looking at the history of that line, I see the reasoning was described in the commit message (17b0bd7713):
```
    * improve  some of the "perl equivalent" entries; for example
      av_store() is *not* like $myarray[$key] = $val, since it replaces the
      stored SV with a different SV, rather than just updating the current
      SV's value.
```

So I guess leave them both using `splice` as the equivalent for now, I'll think later whether it benefits anyone to add more nuance.

Reply:
I've not made any changes here.